src/wallet3/decoy_selection/decoy_selection.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <vector>
+#include "../daemon_comms.hpp"
+#include "../output.hpp"
+#include "../decoy.hpp"
+
+namespace wallet
+{
+  // DecoySelector will choose some a subset of outputs from the provided list of outputs according
+  // to the decoy selection algorithm. The decoys selected should hide the selected output within a 
+  // ring signature and requires careful selection to avoid privacy decreasing analysis
+
+  class DecoySelector
+  {
+   public:
+    std::vector<Decoy>
```

Comment:
instead of fetching the decoys in here should we just return a vector of indecies?

Reply:
Yeah agree, saves me passing around the daemon comms object also